lib/express-routes.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+var _ = require('underscore');
+var express = require('express');
+var bodyParser = require('body-parser');
+var jobs = require('./job-handlers');
+var paths = require('./path-handlers');
+var prepares = require('./prepare-handlers');
+var logger = require('log4js').getLogger('juttle-express-router');
+
+var read_config = require('juttle/lib/config/read-config');
+var JuttleEngineErrors = require('./errors');
+
+var API_PREFIX = '/api/v0';
+
+function default_error_handler(err, req, res, next) {
+    logger.debug('got error "' + err.message + '"');
+
+    // Errors from bodyParser.json() might appear here. Transform
+    // them into the error formats we expect.
+    if (err.message.startsWith('Unexpected token')) {
+        err = JuttleEngineErrors.bundleError(err.message, err.body);
+    } else if (! JuttleEngineErrors.is_juttled_error(err)) {
+        // This error isn't one of the standard errors in
+        // JuttledErrors. Wrap the error in an UnknownError.
+        err = JuttleEngineErrors.unknownError(err);
+    }
+
+    res.status(err.status).send(err);
+}
+
+function add_routes(app, options) {
+
+    let config = read_config(_.pick(options, 'config_path'));
```

Comment:
Oh - on second look it seems the config is actually read twice... all the more reason to pass it in and not read it again.